modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DistributionZoneManager.java

```diff
@@ -985,22 +997,23 @@ private CompletableFuture<Void> scheduleTimers(CatalogZoneDescriptor zone, boole
 
             if (nodesRemoved) {
                 if (zone.consistencyMode() == HIGH_AVAILABILITY) {
-                    if (partitionReset == IMMEDIATE_TIMER_VALUE) {
-                        futures.add(
-                                // TODO: IGNITE-23599 Implement valid behaviour here.
-                                nullCompletedFuture()
-                        );
-                    }
-
                     if (partitionReset != INFINITE_TIMER_VALUE) {
                         zonesState.get(zoneId).reschedulePartitionDistributionReset(
                                 partitionReset,
-                                // TODO: IGNITE-23599 Implement valid behaviour here.
-                                () -> {},
-                                zoneId
+                                () -> {
+                                    fireEvent(
```

Comment:
I'd rather hide it intro private method.

Reply:
Extracted to the method